backend/resources/org/akvo/dash/migrations_tenants/000-tardis.up.sql

```diff
@@ -0,0 +1,75 @@
+CREATE SCHEMA history;
+-- ;;
+
+
+CREATE FUNCTION history.log_change() RETURNS trigger AS $_$
+    DECLARE
+      c refcursor;
+      tt tstzrange;
+    BEGIN
+        IF TG_OP = 'INSERT' THEN
+            EXECUTE 'INSERT INTO history.' || TG_TABLE_NAME ||
+              ' SELECT $1.*, tstzrange(now(), $$infinity$$, $$[)$$)' USING NEW;
+            RETURN NEW;
+        ELSIF TG_OP = 'UPDATE' THEN
+            OPEN c FOR EXECUTE 'SELECT _validrange FROM history.' || TG_TABLE_NAME ||
+              ' WHERE id = $1 ORDER BY _validrange DESC LIMIT 1 FOR UPDATE'
+              USING NEW.id;
+            FETCH FROM c INTO tt;
+
+            IF isempty(tstzrange(lower(tt), now(), $$[)$$)) THEN
+		EXECUTE 'DELETE FROM history.' || TG_TABLE_NAME ||
+		  ' WHERE CURRENT OF ' || quote_ident(c::text);
+            ELSE
+                EXECUTE 'UPDATE history.' || TG_TABLE_NAME || ' SET _validrange = tstzrange($1, now(), $$[)$$)' ||
```

Comment:
This line and the next are indented with tabs, so in my editor they're out of whack. Same in migrations_tenant_manager/000-tardis.up.sql.